src/seedu/addressbook/commands/Command.java

```diff
@@ -43,6 +43,11 @@ public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
 
+    /**
+     * Returns true for command types that mutate the data.
```

Comment:
Try to be more precise here. Some command types mutate the data, but not always. For example, adding a duplicate person won't mutate the data, even though AddCommand is of mutating type. 

A better phrasing would be: Returns true for command types that potentially mutate the data.